From 6648e5d990b0e5932a2996d719431b7b1e819517 Mon Sep 17 00:00:00 2001
From: Alex Williamson <alex.williamson@redhat.com>
Date: Fri, 1 Aug 2014 18:12:54 -0500
Subject: [CHANGE 3/5] Allow prefix to specify a PCI autoboot device location
To: rhvirt-patches@redhat.com,
    jen@redhat.com

Message-id: <20140513195631.6596.23570.stgit@bling.home>
Patchwork-id: 58834
O-Subject: [RHEL6.6 gpxe PATCH 3/5] [prefix] Allow prefix to specify a PCI autoboot device location
Bugzilla: 1057249
RH-Acked-by: Jeffrey Cody <jcody@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

Bugzilla: 1057249
Upstream commit: 90fc273b2b099df4a2c2ca06e1eaa7508e9734be (ipxe)

Signed-off-by: Alex Williamson <alex.williamson@redhat.com>
Modified-by: Michael Brown <mcb30@ipxe.org>
Signed-off-by: Michael Brown <mcb30@ipxe.org>
Signed-off-by: jen <jen@redhat.com>
---
 src/arch/i386/core/pci_autoboot.c | 42 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)
 create mode 100644 src/arch/i386/core/pci_autoboot.c

diff --git a/src/arch/i386/core/pci_autoboot.c b/src/arch/i386/core/pci_autoboot.c
new file mode 100644
index 0000000..9d7a1e9
--- /dev/null
+++ b/src/arch/i386/core/pci_autoboot.c
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2014 Red Hat Inc.
+ *	Alex Williamson <alex.williamson@redhat.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include <stdint.h>
+#include <gpxe/device.h>
+#include <gpxe/init.h>
+#include <realmode.h>
+#include <usr/autoboot.h>
+
+uint16_t __bss16 ( autoboot_busdevfn );
+#define autoboot_busdevfn __use_data16 ( autoboot_busdevfn )
+
+/**
+ * Initialise PCI autoboot device
+ */
+static void pci_autoboot_init ( void ) {
+
+	autoboot_device.bus_type = BUS_TYPE_PCI;
+	autoboot_device.location = autoboot_busdevfn;
+}
+
+/** PCI autoboot device initialisation function */
+struct init_fn pci_autoboot_init_fn __init_fn ( INIT_NORMAL ) = {
+	.initialise = pci_autoboot_init,
+};
-- 
1.9.3

